save-demo/src/main/kotlin/com/saveourtool/save/demo/runners/RunnerFactory.kt

```diff
@@ -0,0 +1,39 @@
+package com.saveourtool.save.demo.runners
+
+import com.saveourtool.save.demo.entity.Demo
+
+/**
+ * Base interface for factory to create [Runner] depending on requested [RunnerType]
+ */
+interface RunnerFactory {
+    /**
+     * @param demo demo entity
+     * @param version version that should be used for demo run
+     * @param type
+     * @return [Runner] created for provided values
+     */
+    fun create(demo: Demo, version: String, type: RunnerType): Runner
+
+    /**
+     * Enum that represents the type of runner that [RunnerFactory] should create
+     */
+    enum class RunnerType {
+        /**
+         * Runner that executes [Demo] in another process
+         */
+        CLI,
+
+        /**
+         * Runner that executes [Demo] in another pod
+         */
+        POD,
+
+        /**
+         * Runner that executes [Demo] in the same process as save-demo
+         *
+         * **Should not be used**
```

Comment:
executes [Demo] in the same process, but shouldn't be used
may be describe why? or remove it at all? we hold this for back compatibility or what?

Reply:
The point is that if we once decide to move `save-demo-cpg` to `save-demo`, it might be useful for some time, until it is started in external pod. 


Will update kdoc and say that this runner might be bad for `save-demo` performance

Reply:
Done